### PR TITLE
[MGS] Add endpoints to get/set a component's active firmware slot

### PR DESCRIPTION
**Pull request by jgallagher**

### CODE REVIEW

**Review by andrewjstone — APPROVED** [review]
Ship it!

File: gateway/src/http_entrypoints.rs [hunk 1]
```diff
@@ -297,6 +297,15 @@ enum PowerState {
     A2,
 }
 
+/// Identifier for an SP's component's firmware slot; e.g., slots 0 and 1 for
+/// the host boom flash.
```
Comment (andrewjstone) [inline gateway/src/http_entrypoints.rs:301]: Just noticed this typo. Presumably you meant 'host boot rom`

File: gateway-cli/src/main.rs [hunk 2]
```diff
@@ -336,8 +337,25 @@ async fn main() -> Result<()> {
         Command::IgnitionCommand { sp, command } => {
             client.ignition_command(sp.type_, sp.slot, command).await?;
         }
-        Command::ComponentActiveSlot { .. }
-        | Command::StartupOptions { .. } => {
+        Command::ComponentActiveSlot { sp, component, set_slot } => {
+            if let Some(slot) = set_slot {
+                client
+                    .sp_component_active_slot_set(
+                        sp.type_,
```
Comment (zephraph) [inline gateway-cli/src/main.rs:344]: What's the significance of the trailing `_` on type?

Comment (jgallagher) [reply gateway-cli/src/main.rs:344]: The OpenAPI spec [says this should be `type`](https://github.com/oxidecomputer/omicron/blob/8dc98ac9e6d6fd6f0b177909af7895060eef7f9b/openapi/gateway.json#L975-L977), but `type` is a reserved word, so progenitor [appends an _](https://github.com/oxidecomputer/progenitor/blob/32f2089f4100613a60dd0bc72674e37f0a72966a/progenitor-impl/src/util.rs#L116-L121) when defining the field name.

File: gateway/src/http_entrypoints.rs (hunk 1 above)
Comment (jgallagher) [reply gateway/src/http_entrypoints.rs]: I do! Thanks, fixed.

**Pull request merged**